### PR TITLE
py38: upgrade to DRF 3.11.x

### PR DESCRIPTION
**Pull request by joshuarli**
DRF 3.11 supports Python 3.8. Since 3.8 there aren't any interesting changes, most of it is openapi schema generation stuff which we don't use. Should just be an easy bump I'm hoping.

I also resynced the monkeypatch for DRF 3.8.x.

### CODE REVIEW

**Comment by joshuarli** [comment]
Hmmm pretty stumped on this:

```
tests/snuba/api/endpoints/test_organization_events_histogram.py F                                                                                             [100%]

============================================================================= FAILURES ==============================================================================
______________________________________________ OrganizationEventsHistogramEndpointTest.test_bad_params_missing_fields _______________________________________________
tests/snuba/api/endpoints/test_organization_events_histogram.py:129: in test_bad_params_missing_fields
    assert response.data == {
E   AssertionError: assert {'field': [Er...='required')]} == {'field': ['T...t be empty.']}
E     Differing items:
E     {'field': [ErrorDetail(string='This field is required.', code='required')]} != {'field': ['This list may not be empty.']}
E     Use -v to get the full diff
```

I narrowed it down to 3.9.0 and there's nothing that seems interesting in https://github.com/encode/django-rest-framework/compare/3.8.2...3.9.0...

File: tests/snuba/api/endpoints/test_organization_events_histogram.py [hunk 1]
```diff
@@ -127,7 +128,7 @@ def test_bad_params_missing_fields(self):
         response = self.do_request(query)
         assert response.status_code == 400
         assert response.data == {
-            "field": ["This list may not be empty."],
+            "field": [ErrorDetail(string="This field is required.", code="required")],
```
Comment (wedamija) [inline tests/snuba/api/endpoints/test_organization_events_histogram.py:131]: @Zylphrex I saw that you wrote this code originally, do you know if the frontend is relying on this specific value?

**Review by wedamija — COMMENTED** [review]
Looks ok to me, just want to get @Zylphrex's feedback

File: tests/snuba/api/endpoints/test_organization_events_histogram.py (hunk 1 above)
Comment (Zylphrex) [reply tests/snuba/api/endpoints/test_organization_events_histogram.py:131]: Can confirm that the frontend does not depend on the specific value.

**Review by wedamija — APPROVED** [review]
lgtm, changelog looks good and changes seem minor for the functionality we use.

**Pull request merged**